<launch>
    <!-- world and robot -->
    <!-- arg for robots -->
    <arg name="x" default="0.0"/>
    <arg name="y" default="0.0"/>
    <arg name="z" default="0.0"/>
    <arg name="roll" default="0.0"/>
    <arg name="pitch" default="0.0"/>
    <arg name="yaw" default="0.0"/>


    <!-- world -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
    	<arg name="world_name" value="$(find servebot_project)/worlds/servebot_cafe.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!-- spawn robot on world -->
    <arg name="model" default="burger" doc="model type [burger, waffle]" />
    <param name="robot_description" command="$(find xacro)/xacro $(find servebot_project)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model turtlebot3_burger -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg roll) -P $(arg pitch) -Y $(arg yaw) -param robot_description" />
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/>
    <node name="robot_state_publisher_turtlebot3" pkg="robot_state_publisher" type="robot_state_publisher"
        respawn="true" output="screen">
        <param name="publish_frequency" type="double" value="1.5" />
    </node>





    <!-- 20240122 추가내용 -->
    
    <!-- setting tf from map to odom -->
    <!-- args를 0 0 0 0 0 0 에 두면 -->
    <node pkg="tf" type="static_transform_publisher" name="map_to_odom_publisher"
    args="0 0 0 0 0 0 map odom 30" />
    <!-- args="2.0 0.5 0 0 0 0 map odom 30" /> -->
    
    <!-- serving static map -->
    <node pkg="map_server" type="map_server" name="map_server" args="$(find servebot_project)/map/map.yaml"/>
    
    <!-- 20240125 amcl 추가 (에러발생) 중복된TF받아옴.-->
    <!-- AMCL.launch.xml -->
    <include file="$(find servebot_project)/launch/amcl.launch.xml" />
    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find servebot_project)/rviz/serve.rviz"/>
    
    <!-- move_base -->
    <arg name="cmd_vel_topic" default="/cmd_vel" />
    <arg name="odom_topic" default="odom" />
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <param name="base_global_planner" value="navfn/NavfnROS" />

    <rosparam file="$(find servebot_project)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find servebot_project)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find servebot_project)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find servebot_project)/param/global_costmap_params.yaml" command="load" />
    <!-- <rosparam file="$(find servebot_project)/param/global_costmap_params_odom.yaml" command="load"/> -->
    <rosparam file="$(find servebot_project)/param/move_base_params.yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
  </node>
    
</launch>